src/core/NewDomain.ml

```diff
@@ -363,8 +366,10 @@ struct
     | Tm.Next _ ->
       raise CanJonHelpMe
 
-    | Tm.Data _ ->
-      raise CanJonHelpMe
+    | Tm.Data info ->
+      let params = List.map (fun t -> Val (Delayed.make @@ lazy (eval rel env t))) info.params in
```

Comment:
@favonia Is this the way i'm supposed to write this code? Or is something else preferred?